asset-managment/niapm.yml

```diff
@@ -0,0 +1,1405 @@
+      - PAST_RECOMMENDED_DUE_DATE
+  CalibrationMode:
+    title: Calibration Mode
+    description: Specifies how calibration data has been recoded; either automatically discovered or manually entered.
+    type: string
+    enum:
+      - AUTOMATIC
+      - MANUAL
+  CalibrationType:
+    title: Calibration Type
+    description: Specifies the type of the calibration; either self or external calibration.
+    type: string
+    enum:
+      - SELF_CALIBRATION
+      - EXTERNAL_CALIBRATION
+  AssetState:
+    title: Asset State
+    description: Specifies the state of the asset and its system.
+    type: object
+    properties:
+      systemConnectionState:
+        $ref: '#/definitions/SystemConnectionState'
+      assetPresence:
+        $ref: '#/definitions/AssetPresenceStatus'
+  TemperatureSensor:
+    title: Temperature Sensor
+    description: Specifies the sensor name and temperature reading, in Celsius.
+    type: object
+    required:
+        - reading
+    properties:
+      name:
+        description: Sensor name
+        type: string
+        example: Sensor0
+      reading:
+        description: Sensor reading
+        type: number
+        format: double
+        example: 25.8
+  AssetLocation:
+    title: Asset Location
+    description: Specifies information about the asset and the system in which the asset resides.
+    type: object
+    properties:
+      minionId:
+        description: Id of the minion the asset is located in.
+        type: string
+        example: NI_PXIe-8135_Embedded_Controller--MAC-00-80-2F-23-52-65
+      parent:
+        description: The parent of the asset.
+        type: string
+        example: Chassis1
+      resourceUri:
+        description: Identifier of a resource.
+        type: string
+        example: 1/4243/1949942980/01BB877A/3
+      slotNumber:
+        description: Number of the slot the asset is located in.
+        type: integer
+        example: 2
+      systemName:
+        description: Hostname of the system.
+        type: string
+        example: APM-PXI1
+      state:
+        $ref: '#/definitions/AssetState'
+  ExternalCalibrationOperator:
+    title: External Calibration Operator
+    description: Specifies the operator that performed the calibration.
+    type: object
+    properties:
+      displayName:
+        description: Display name of logged in user.
+        type: string
+      userId:
+        description: Identifier of the logged in user.
+        type: string
+  SelfCalibrationModel:
+    title: Self Calibration
+    description: Contains data from the last self calibration.
+    type: object
+    required:
+        - date
+    properties:
+      temperatureSensors:
+          description: An array of temperature sensor information.
+          type: array
+          items:
+            $ref: '#/definitions/TemperatureSensor'
+      isLimited:
+        description: Specifies whether the last internal/self calibration of the asset was a limited calibration.
```

Comment:
Are internal and self-calibration one and the same? If so, could we use only self-calibration since we haven't mentioned internal before now? I see that it's used a bit in some of the descriptions below, but let's just use self-calibration vs. introducing other variations of it.